### PR TITLE
Homework CMake/tests

### PR DESCRIPTION
**Pull request by Ruthietta**
Authors: @mariuszlisowski @Petrus9107 @m-kop @dawidjozefczak @Marimeiko @ms00780

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ on: [push, pull_request]
 
 jobs:
   checker:
+  SHM:
```
Comment (Kasperice) [inline .github/workflows/main.yml]: Will this work correctly? You can specify two jobs with different names, but now you will have (I assume) empty "checker" and "SHM" which will do everything

**Review by Kasperice — COMMENTED** [review]
As I see checks are not running. I had to do `git pull --rebase coders master` then push branch again. Then "checks" were working. Then you will also see if your main.yml file is correct or not

File: .github/workflows/main.yml (hunk 1 above)
Comment (ziobron) [reply .github/workflows/main.yml]: Yes, this is problematic. Merged unsuccessfully. Do not mix jobs.

File: shm/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+# later. Otherwise we have to add them here ourselves.
+if (CMAKE_VERSION VERSION_LESS 2.8.11)
+  include_directories("${gtest_SOURCE_DIR}/include")
+endif()
+
+##-----------------G-test SCRIPT FROM CODERS SCHOOL----------------##
+
+set(PROJECT_NAME SHM)
+
+project(${PROJECT_NAME})
```
Comment (ziobron) [inline shm/CMakeLists.txt]: Replace with

```suggestion
project(SHM)
```

The effect is the same. `project` function set `PROJECT_NAME` variable automatically.

File: shm/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+enable_testing()
+add_definitions(-D Wall -Wextra -Werror -pedantic)
```
Comment (ziobron) [inline shm/CMakeLists.txt]: please use `target_compile_options` to set compiler flags. BTW you have a mistake here - there is no - before Wall

File: shm/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+)
+
+add_library(${CMAKE_PROJECT_NAME}-lib STATIC ${SOURCES})
+
+add_executable(${CMAKE_PROJECT_NAME} ${SOURCE_DIR}/main.cpp)
+add_executable(${CMAKE_PROJECT_NAME}-ut ${TEST_DIR}/test.cpp)
+
+target_link_libraries(${CMAKE_PROJECT_NAME} ${CMAKE_PROJECT_NAME}-lib)
+target_link_libraries(${CMAKE_PROJECT_NAME}-ut ${CMAKE_PROJECT_NAME}-lib gtest_main)
+add_test(NAME shm_test COMMAND ${CMAKE_PROJECT_NAME}-ut)
```
Comment (ziobron) [inline shm/CMakeLists.txt]: Please use ${PROJECT_NAME}. ${CMAKE_PROJECT_NAME} is not bad, however ${PROJECT_NAME} is more often used and know.

**Review by ziobron — CHANGES_REQUESTED** [review]

**Review by ziobron — APPROVED** [review]

**Comment by ziobron** [comment]
Great, everything is good now :)

10 x 2XP for each point in cmake homework (2767)
2 x 2XP for timely delivery (1119)

{
  "requirement_ids": [2767, 2767, 2767, 2767, 2767, 2767, 2767, 2767, 2767, 2767, 1119, 1119],
  "users": ["mariuszlisowski" , "Petrus9107", "m-kop", "dawidjozefczak", "Marimeiko", "ms00780"]
}

**Pull request closed without merging**